--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,8 @@
-cmake_minimum_required(VERSION 3.16.0 FATAL_ERROR) # Configurable policies: <= CMP0097
+cmake_minimum_required(VERSION 3.25.0 FATAL_ERROR) # Configurable policies: <= CMP0097
 
 cmake_policy(SET CMP0091 NEW)
 cmake_policy(SET CMP0092 NEW)
+cmake_policy(SET CMP0141 NEW)
 
 list(APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake/Modules)
 include(LibtorrentMacros)
@@ -13,6 +14,7 @@
 	VERSION ${VER_MAJOR}.${VER_MINOR}.${VER_TINY}
 )
 set (SOVERSION "${VER_MAJOR}.${VER_MINOR}")
+set (CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "Embedded")
 
 include(GNUInstallDirs)
 include(GeneratePkgConfig)
@@ -573,6 +575,9 @@
 		PRIVATE TORRENT_BUILDING_SHARED
 		INTERFACE TORRENT_LINKING_SHARED
 	)
+else()
+	set(Boost_USE_STATIC_LIBS ON)
+	set(OPENSSL_USE_STATIC_LIBS OFF)
 endif()
 
 set_target_properties(torrent-rasterbar
@@ -614,6 +619,8 @@
 
 # Unconditional platform-specific settings
 if (WIN32)
+	set (CMAKE_MSVC_DEBUG_INFORMATION_FORMAT "Embedded")
+
 	target_link_libraries(torrent-rasterbar
 		PUBLIC
 			bcrypt mswsock ws2_32 iphlpapi
@@ -639,11 +646,13 @@
 				# https://learn.microsoft.com/en-us/cpp/build/reference/permissive-standards-conformance?view=msvc-170
 				/permissive-
 				# https://docs.microsoft.com/en-us/cpp/build/reference/utf-8-set-source-and-executable-character-sets-to-utf-8?view=msvc-170
-				/utf-8
-				# https://devblogs.microsoft.com/cppblog/msvc-now-correctly-reports-__cplusplus/
 				/Zc:__cplusplus
+				$<$<NOT:$<CONFIG:Debug>>:/GL> # /LTCG counterpart
+		)
+		target_link_options(torrent-rasterbar
+			PRIVATE
+				$<$<NOT:$<CONFIG:Debug>>:/LTCG /OPT:ICF=5 /OPT:REF>
 		)
-		set_target_properties(torrent-rasterbar PROPERTIES LINK_FLAGS_RELEASE "/OPT:ICF=5 /OPT:REF")
 	endif()
 endif()
 
